Upgrade checkstyle

alanorth opened the pull request:
## References
* Fixes #8441

## Description
Upgrade checkstyle to the latest stable upstream version.

## Instructions for Reviewers

List of changes in this PR:
* Upgrade checkstyle from version 8.45.1 to 10.19.0 (see [checkstyle release notes](https://checkstyle.org/releasenotes.html))
* Fix one "GenericWhitespace" warning

Try running checkstyle, ie:

```console
$ mvn -U checkstyle:check
```

alanorth committed:
pom.xml: update checkstyle

Version 8.45.1 to 10.19.0.

mwoodiupui reviewed (commented):
This looks good.  I checked out the branch and ran 'checkstyle:check'.  It worked.  Unfortunately it found 60 errors in 'dspace-server-webapp'.  It's also been subjected to a merge conflict.

alanorth commented:
Thanks @mwoodiupui. I dropped the conflicting `maven-checkstyle-plugin` version bump since it was handled in another large dependency update. I also bumped to latest checkstyle 10.13.0.

We still have the issue with indentation of annotation children (see PR notes).

alanorth committed:
dspace-api: fix GenericWhitespace warning

Checkstyle was complaining:

    (whitespace) GenericWhitespace: '>' is followed by whitespace.

tdonohue commented:
NOTE: this conflict is from #9940, which updated us to the latest version of Checkstyle 8.x.  So, I suspect you may be able to *remove* some of the changes in this PR, as they now already exist on `main`.

alanorth commented:
Thanks @tdonohue. I updated the PR to use the latest checkstyle and drop unneeded changes already applied as you suggested.

tdonohue reviewed (approved):
👍 Thanks @alanorth !  This is obviously working well (passes all checkstyle checks & automated tests), so I'll merge immediately.  For now, I'm *not* going to backport this to 8.x/7.x (as that'd likely require manual backports, and I'm not sure this is important to backport at this time).  We can always backport later if we find a need to do so.
